src/data/apigateway/mod.rs

```diff
@@ -0,0 +1,170 @@
+#[cfg(test)]
+mod tests;
+
+pub mod auth;
+
+use chrono::prelude::*;
+use std::fmt;
+use std::collections::HashMap;
+use serde;
+use serde_qs as qs;
+
+#[derive(Debug,Eq,PartialEq,Serialize,Deserialize)]
+pub enum HttpMethod {
+    HEAD,
+    GET,
+    POST,
+    PUT,
+    OPTIONS,
+    DELETE,
+}
+
+impl fmt::Display for HttpMethod {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", match *self {
+            HttpMethod::HEAD => "HEAD",
+            HttpMethod::GET => "GET",
+            HttpMethod::POST => "POST",
+            HttpMethod::PUT => "PUT",
+            HttpMethod::OPTIONS => "OPTIONS",
+            HttpMethod::DELETE => "DELETE"
+        })
+    }
+}
+
+#[derive(Serialize,Deserialize)]
+pub struct HttpEvent {
+    pub body: Option<String>,
+    pub headers: Option<HashMap<String, String>>,
+    #[serde(rename="httpMethod")]
+    pub http_method: HttpMethod,
+    #[serde(rename="isBase64Encoded")]
+    pub is_base64_encoded: bool,
+    pub path: String,
+    #[serde(rename="pathParameters")]
+    pub path_parameters: Option<HashMap<String, String>>,
+    #[serde(rename="queryStringParameters")]
+    pub query_string_parameters: Option<HashMap<String, String>>,
+    pub resource: String,
+    #[serde(rename="requestContext")]
+    pub request_context: HttpEventRequestContext,
+    #[serde(rename="stageVariables")]
+    pub stage_variables: Option<HashMap<String, String>>,
+}
+
+impl HttpEvent {
+
+    pub fn get_header(&self, key: &str) -> Option<&str> {
+        match self.headers {
+            Some(ref h) => h.get(key).map(|s| s.as_str()),
+            None => None,
+        }
+    }
+}
+
+impl fmt::Display for HttpEvent {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{protocol} {method} {path}{querystring} (body? {body})",
+            protocol = match self.request_context.protocol {
+                Some(ref s) => s.as_str(),
+                None => "(unknown)"
+            },
+            method = self.http_method,
+            path = self.path,
+            querystring = qs::to_string(&self.query_string_parameters).unwrap_or(String::new()),
+            body = match self.body {
+                Some(_) => true,
+                None => false
+            }
+        )
+    }
+}
+
+#[derive(Serialize,Deserialize)]
+pub struct HttpEventRequestContext {
```

Comment:
I think you can largely get a way with a struct level `#[serde(rename_all = "camelCase")]` here

Reply:
note: comments like this are not really blockers for me and could easily fit into followups